### ISSUE TITLE
Typescript definitions incomplete

### ISSUE THREAD

**Issue by MrMasterplan** [mapeditor/tiled#3038]
I noticed the typescript definitions are incomplete:
https://github.com/mapeditor/tiled/blob/068d0bf2e9632bf7473b40e4073eb3787f8889d1/docs/scripting-doc/index.d.ts
They are missing some things I need such as Asset.isTileset, Object.setProperty, all the details of Tileset, just to name a few.
If you are open to contributions I would be willing to contribute these in a pull request. Just let me know if I can add them directly to that file or if the file is generated in some way from other code.

**Comment by bjorn** [comment]
Ah, this work is still WIP at #2965. Since @Ablu does not appear to have time to finish it maybe I should just merge it and then you could help with any rough edges?

Also, apart from providing code completion, one of the goals of the TypeScript definitions was also to replace the currently manually edited documentation. But I'm afraid that after the recent website redesign, the styling of these pages is completely off (https://www.mapeditor.org/docs/scripting/). I'll look into this soon.

**Comment by MrMasterplan** [comment]
Like I said, I would be happy to give it a shot. For me the missing definitions are a pain point at the moment.

**Comment by bjorn** [comment]
@MrMasterplan I've published a new `1.4.3-beta` version of `tiled-api` which has a much more complete and documented version of the API (mainly missing stuff added in Tiled 1.5). Any help in improving this package further is very appreciated!

**Comment by MrMasterplan** [comment]
I added a lot to definitions. See the PR here: https://github.com/mapeditor/tiled/pull/3040

**Comment by MrMasterplan** [comment]
I am finding more omissions as I go along. Hold out a bit on publishing to npm. I will add another PR soon.

**Comment by MrMasterplan** [comment]
Added another PR https://github.com/mapeditor/tiled/pull/3041 
There are still some skeleton definitions, but for me this didn't matter. I can now use the file. Once the PR is merged, please report back when the definitions have been published to npm. I will hold out on publishing my project rwklevelfiles until this dependency is available in npm.

**Comment by MrMasterplan** [comment]
Thank you!!

**Comment by bjorn** [comment]
Today I've completed the migration of the documentation to the `index.d.ts` file. It should now be at least as complete as the reference in the manual used to be. The reference has been removed from [the latest version of the manual](https://doc.mapeditor.org/en/latest/reference/scripting/).

The search is also functional in the generated documentation now.

I've released `tiled-api` 1.6.0 (no longer in beta).
